jmxtrans-core/src/main/java/com/googlecode/jmxtrans/executors/SeparateExecutorRepository.java

```diff
@@ -0,0 +1,68 @@
+/**
+ * The MIT License
+ * Copyright © 2010 JmxTrans team
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package com.googlecode.jmxtrans.executors;
+
+import com.googlecode.jmxtrans.model.Server;
+import com.googlecode.jmxtrans.monitoring.ManagedThreadPoolExecutor;
+
+import javax.annotation.Nonnull;
+import javax.management.MalformedObjectNameException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ThreadPoolExecutor;
+
+public class SeparateExecutorRepository implements ExecutorRepository {
+	@Nonnull private final ConcurrentHashMap<Server, ThreadPoolExecutor> repository;
+	@Nonnull private final ExecutorFactory executorFactory;
+	@Nonnull private final ArrayList<ManagedThreadPoolExecutor> mBeans;
```

Comment:
mBeans should probably be thread safe. We are good accidentally, since initialization and shutdown are done in the same thread, but that's accidentally (I suspect the failing case is an edge case we'll never see, but better be safe than sorry). This means either making it a concurrent list (CopyOnWriteArrayList?) or immutable and move the initialization outside of the class. Or synchronize all access to mBeans...

Side note: the variable should be declared as `List`.

Reply:
You probably right. I'll make some changes.